docs/design/datacontracts/RuntimeTypeSystem.md

```diff
@@ -137,6 +137,31 @@ partial interface IRuntimeTypeSystem : IContract
     // Return true if a MethodDesc represents an IL Stub dynamically generated by the runtime
     // A IL Stub method is also a StoredSigMethodDesc, and a NoMetadataMethod
     public virtual bool IsILStub(MethodDescHandle methodDesc);
+
+    // Return true if a MethodDesc is in a collectible module
+    public virtual bool IsCollectibleMethod(MethodDescHandle methodDesc);
+
+    // Return true if a MethodDesc is in a module that supports Edit-And-Continue
+    public virtual bool InEnCEnabledModule(MethodDescHandle methodDesc);
+
+    // Return true if a MethodDesc supports mulitiple code versions
+    public virtual bool IsVersionable(MethodDescHandle methodDesc);
+
+    // Return a pointer to the IL versioning state of the MethodDesc
+    public virtual TargetPointer GetMethodDescVersioningState(MethodDescHandle methodDesc);
```

Comment:
We're going to need the ability to walk the list of IL versions, and Native code versions, and query various details about these things. This doesn't feel like the right api. Could you get it onto the CodeVersions contract?

Reply:
This just extracts the address from the MethodDescCodeData.  All the logic for working with the versioning state is in CodeVersions.

It's not that different from 

```c
  TargetPointer modAddr =  RuntimeTypeSystem.GetModule(TypeHandle th);
  ModuleHandle mod = Loader.GetModuleHandle(modAddr);
```

RTS gives you a pointer and Loader turns it into a handle that you can use for queries.  Similarly RTS will give a pointer to the versioning state and CodeVersions will consume it in some manner.   

Currently the versioning state is consumed entirely internally in the CodeVersions contract, but in the future it could be exposed via a handle or through some data API  (see `GetSpecificNativeCodeVersion` and `IsActiveNativeCodeVersion` implementations)